net: openthread: allow to enable/disable TCP

edmont opened the pull request:
Allow to enable/disable the OpenThread TCP implementation. Disable it by default.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

pdgendt commented on subsys/net/lib/openthread/platform/openthread-core-zephyr-config.h:
```diff
@@ -401,4 +401,24 @@
+ * Enable TCP.
+ *
+ */
+#ifdef CONFIG_OPENTHREAD_TCP_ENABLE
+#define OPENTHREAD_CONFIG_TCP_ENABLE CONFIG_OPENTHREAD_TCP_ENABLE
+#endif /* CONFIG_OPENTHREAD_TCP_ENABLE */
```
The default in `openthread` is to [enable it](https://github.com/zephyrproject-rtos/openthread/blob/zephyr/src/core/config/ip6.h#L186).

Maybe always define it?
```C
#define OPENTHREAD_CONFIG_TCP_ENABLE IS_ENABLED(CONFIG_OPENTHREAD_TCP_ENABLE)
```

edmont replied:
Agree. Same goes for `OPENTHREAD_CONFIG_CLI_TCP_ENABLE`.

pdgendt commented on subsys/net/l2/openthread/Kconfig.thread:
```diff
@@ -144,3 +144,10 @@ config OPENTHREAD_IP6_MAX_EXT_MCAST_ADDRS
+config OPENTHREAD_TCP_ENABLE
+	bool "Enable TCP"
+
+config OPENTHREAD_CLI_TCP_ENABLE
+	bool "Enable TCP in the CLI tool"
+	default y if OPENTHREAD_TCP_ENABLE
```
```suggestion
config OPENTHREAD_CLI_TCP_ENABLE
	bool "Enable TCP in the CLI tool"
	default y if SHELL
	depends on OPENTHREAD_TCP_ENABLE
```

edmont replied:
Done.

edmont committed:
net: openthread: allow to enable/disable TCP

Allow to enable/disable the OpenThread TCP implementation. Disable
it by default.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
